sdk/storage/azure-storage-common/src/file_io.cpp

```diff
@@ -155,15 +155,13 @@ namespace Azure { namespace Storage { namespace _internal {
 
   FileWriter::~FileWriter() { close(m_handle); }
 
-  void FileWriter::Write(const uint8_t* buffer, int64_t length, int64_t offset)
```

Comment:
@Jinming-Hu can the offset be made `size_t` as well?

Also, can you take over this branch/PR and make the changes to the callsite as part of https://github.com/Azure/azure-sdk-for-cpp/issues/2313?

For example, it requires changes here:
https://github.com/Azure/azure-sdk-for-cpp/blob/e5763720d4a08a90b2e5e63525957e5051387430/sdk/storage/azure-storage-files-shares/src/share_file_client.cpp#L802-L815

Reply:
> can the offset be made size_t as well?

Yes, I think so.

> Also, can you take over this branch/PR and make the changes to the callsite as part of #2313?

Do you want to me append commits to this PR or open a new PR?

Reply:
Whatever's easier for you. Create a new PR, and I will close this one. :)